### ISSUE TITLE
v6.5.0-rc5 seems to be much slower than v6.4.3 with the big torrent file

### ISSUE THREAD

**Issue by vi** [Tribler/tribler#1689]
With [this](http://www.tlmc.eu/) torrent in Download list, Tribler v6.4.3 was very slow at startup (several minutes of unresponsive UI), then seriously slow, but yet somewhat usable afterwards.

v6.5.0-rc5 seems to be even slower, rendering it just unusable. It has downloaded a file from it, but UI still shows "Downloading 0.0%" when the UI finally got unblocked.

**Comment by whirm** [comment]
I'll have a shot at it.

I guess that's on Linux, right?

**Comment by whirm** [comment]
Is that torrent copyleft/public domain?

**Comment by vi** [comment]
Yes, it's on Linux.

> Is that torrent copyleft/public domain?


Unlikely, it's a big collection of related music from multiple sources. 

Note that it's one of biggest torrent files easily available, so it's tricky to download it entirely without preparation.

Maybe the torrent file itself is OK (I don't see any info).

There is also a 7z with cue sheets (just enumeration of all tracks included) inside, so you can download just it avoiding downloading any actual content.

**Comment by whirm** [comment]
@vi You mean that Tribler slows down even without downloading any content from the file?

**Comment by vi** [comment]
Yes.

The torrent is that big (there was a warning on it's site about client incompatibilities).

But I expect Tribler to be a good BitTorrent client and handle tricky cases as well. It should slow down logarithmically depending on indexed content, not linearly. And should not block the UI thread when busy.

**Comment by whirm** [comment]
Hm, I'll have a look post 6.5.

Thanks for the report!

**Comment by devos50** [comment]
Actually, the GUI should not be blocked in Tribler 7 so hopefully, this issue is resolved somewhat

**Comment by devos50** [comment]
I've optimised a bit in https://github.com/Tribler/tribler/commit/a1c707cfb81a718f6aca0796eeaa633e49cb2741 and the files are displayed in a reasonable amount of time. However, after adding the torrent to the libtorrent engine, Tribler slows down dramatically and I think this is due to libtorrent eating all CPU.

I will postpone this issue to somewhere after the Tribler 7 release.

**Comment by ichorid** [comment]
Libtorrent interface refactoring discussed in #3418 should have this problem solved. Will mention this problem in tasklist there.
Closing this one.
